server/src/internalClusterTest/java/org/elasticsearch/indices/TestSystemIndexDescriptorAllowsTemplates.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A special kind of {@link SystemIndexDescriptor} that can toggle what kind of mappings it
+ * expects. A real descriptor is immutable.
+ */
+public class TestSystemIndexDescriptorAllowsTemplates extends TestSystemIndexDescriptor {
```

Comment:
Can you rename this class? If the javadoc is right, the special thing isn't that it allows templates, it's that the mapping is switchable.

Reply:
The javadoc is wrong, I'll fix it. The purpose of the extra descriptor is to be doing exactly the same as the other test descriptor, but also allow templates to apply to it like the kibana-sessions templates.